trade-finance-manager-ui/component-tests/utilisation-reports/_macros/premium-payments-table.component-test.ts

```diff
@@ -137,13 +124,18 @@ describe(component, () => {
     wrapper.expectElement(`${tableSelector} thead th:contains("Total payments received")`).notToHaveAttribute('aria-sort');
   });
 
-  it('should render the select all checkbox in the table headings row when userCanEdit is true', () => {
-    const wrapper = getWrapper();
+  it('should render the select all checkbox in the table headings row when userCanEdit and hasSelectableRows are true', () => {
+    const wrapper = render({ ...defaultRendererParams(), userCanEdit: true, hasSelectableRows: true });
     wrapper.expectElement(`${tableSelector} thead td input[type="checkbox"]#select-all-checkbox`).toExist();
   });
 
   it('should not render the select all checkbox in the table headings row when userCanEdit is false', () => {
-    const wrapper = render({ ...defaultRendererParams(), userCanEdit: false });
+    const wrapper = render({ ...defaultRendererParams(), userCanEdit: false, hasSelectableRows: true });
+    wrapper.expectElement(`${tableSelector} thead th input[type="checkbox"]#select-all-checkbox`).notToExist();
+  });
+
+  it('should not render the select all checkbox in the table headings row when hasSelectableRows is false', () => {
```

Comment:
Slightly picky - should we reference userCanEdit being true in the naming too?

e.g. 
```suggestion
  it('should not render the select all checkbox in the table headings row when userCanEdit is true and hasSelectableRows is false', () => {
```

Happy to hear thoughts, this would also apply to the above test case

Reply:
I left it out to try and keep the name shorter but also precise to the assertion. I have set the userCanEdit to true explicitly so that the test actually fails if you change it, but I don't think we're really too interested in that in this test? What do you think?